stackslib/src/chainstate/burn/operations/stack_stx.rs

```diff
@@ -193,9 +208,26 @@ impl StackStxOp {
         let stacked_ustx = parse_u128_from_be(&data[0..16]).unwrap();
         let num_cycles = data[16];
 
+        let mut signer_key: Option<StacksPublicKeyBuffer> = None;
+        let mut max_amount: Option<u128> = None;
+        let mut auth_id: Option<u32> = None;
+
+        if data.len() >= 50 {
+            signer_key = Some(StacksPublicKeyBuffer::from(&data[17..50]));
+        }
+        if data.len() >= 66 {
```

Comment:
This could be made slightly safer with the following:

```rust
if data.len() >= 66 {
   let Some(amt) = parse_u128_from_be(&data[50..66]) else {
      return None;
   };
   max_amount = Some(amt);
}
if data.len() >= 70 {
   let Some(id) = parse_u32_from_be(&data[66..70]) else {
      return None;
   };
   auth_id = Some(id);
}
```

Reply:
Added 👍🏼